libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m

```diff
@@ -281,21 +282,13 @@ - (void)beginUserAgentFlow {
     } else {
         [approvalUrl appendFormat:@"&%@=%@", kSFOAuthResponseType, kSFOAuthResponseTypeToken];        
     }
-        
-    if ([self.scopes count] > 0) {
-        //append scopes
-        [approvalUrl appendFormat:@"&%@=", kSFOAuthScope];
-        NSMutableString *scopeStr = [[NSMutableString alloc] initWithString:kSFOAuthRefreshToken];
-
-        for (NSString *scope in self.scopes) {
-            if (![scope isEqualToString:kSFOAuthRefreshToken]) {
-            	[scopeStr appendFormat:@" %@", scope]; // scopes are delimited by a space character
-            }
-        }
-        
-        NSString *finalScopeStr = [scopeStr stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-        [approvalUrl appendString:finalScopeStr];
-    }
+    
+    // Adding refresh_token and custom_permissions scopes
+    NSMutableSet* scopes = [NSMutableSet setWithSet:self.scopes];
+    [scopes addObject:kSFOAuthRefreshToken];
+    [scopes addObject:kSFOAuthCustomPermissions];
```

Comment:
We need the custom_permissions scope to get the custom permissions back in the id response